ckanext/cnra_schema/helpers.py

```diff
@@ -14,58 +18,40 @@ def delete_existing_extra_from_package_dict(key, package_dict):
         package_dict['extras'].remove(existing_extra)
 
 
-def set_waf_map_fields(package_dict, iso_values, map_fields):
-    if map_fields:
-        for map_field in map_fields:
-            source_field = map_field.get('source')
-            target_field = map_field.get('target')
-            default_value = map_field.get('default')
-            value = iso_values.get(source_field, default_value)
-
-            # If value is a list, convert to string
-            if isinstance(value, list):
-                value = ', '.join(str(x) for x in value)
-
-            package_dict[target_field] = value
-
-            # Remove from extras any keys present in the config
-            delete_existing_extra_from_package_dict(target_field, package_dict)
-
-    return package_dict
-
-
-def set_waf_publisher_values(package_dict, iso_values, publisher_mapping):
-    publisher_field = publisher_mapping.get('publisher_field')
-
-    if publisher_field:
-        publisher_name = iso_values.get('publisher') or \
-                         publisher_mapping.get('default_publisher')
-        package_dict[publisher_field] = publisher_name
+def get_date_and_time_dict(date_timestamp):
+    if isinstance(date_timestamp, list) and len(date_timestamp) > 0:
+        date_timestamp = date_timestamp[0]
 
-        # Remove from extras any keys present in the config
-        delete_existing_extra_from_package_dict(publisher_field, package_dict)
+    period = {
+        'date': '',
+        'time': ''
+    }
 
-    return package_dict
+    date_timestamp_len = len(date_timestamp)
+    if date_timestamp_len > 10:
```

Comment:
`dateutil.parser` can be used to parse dates and timestamps from different possible formats. It returns a `datetime.datetime` object which can be used to get the date and time separately.